lib/perl/NethServer/ApiTools.pm

```diff
@@ -99,7 +99,12 @@ sub error
 #
 sub readInput
 {
-    my $str = do { local $/; <STDIN> };
+
+    my $tmp = do { local $/; <STDIN> };
+
+    # Make sure everything is UTF-8 encoded
+    use Encode qw(encode);
+    my $str = encode("UTF-8", $tmp);
```

Comment:
what happens if $tmp has a wrong encoding? does Perl die()?

Reply:
No it doesn't die, but we can use CROAK if we want to: https://perldoc.perl.org/Encode.html#Handling-Malformed-Data